src/crypto/hmac.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @fileoverview This module implements an abstracted interface over
+ * HMAC implementation from asmcrypto.js
+ * @module crypto/hmac
+ * @private
+ */
+
+import { Hmac } from '@openpgp/asmcrypto.js/dist_es8/hmac/hmac';
+import { Sha1 } from '@openpgp/asmcrypto.js/dist_es8/hash/sha1/sha1';
+import { Sha256 } from '@openpgp/asmcrypto.js/dist_es8/hash/sha256/sha256';
+import { Sha512 } from '@openpgp/asmcrypto.js/dist_es8/hash/sha512/sha512';
```

Comment:
asmcrypto.js's SHA512 implementation is very fast, but very big. It may not be worth it to import it just for hashing very small values very occasionally.
Tbh, perhaps you could reimplement HMAC, using `crypto.hash.digest`? If all the data is available at once, it should be quite simple, see https://en.wikipedia.org/wiki/HMAC#Implementation.